### PR TITLE
[doc] add AdaScale API doc

### PR DESCRIPTION
**Pull request by min-xu-ai**
- removed experimental warning as we have validated it on cifar and
imagenet, transformer is looking good so far too.
- fixed API doc formatting
- make it consistent with the other code in the repo
- tested by making the doc locally and inspect the results

# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos, doc improvements)
- [x] Did you read the [contributor guideline](https://github.com/facebookresearch/fairscale/blob/master/CONTRIBUTING.md)?
- [x] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?

## What does this PR do?
Fixes # (issue).

## PR review
Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by min-xu-ai** [comment]
@msbaines @Vittorio-Caggiano @blefaudeux, I check the doc build locally. I am wonder it there a way to check the doc build on the PR? Just like we can check the results of circleci. No worries if that's not already available. Not very critical to me, just curious.

File: docs/source/index.rst [hunk 1]
```diff
@@ -1,7 +1,8 @@
 .. fairscale documentation master file, created by
    sphinx-quickstart on Tue Sep  8 16:19:17 2020.
-   You can adapt this file completely to your liking, but it should at least
-   contain the root `toctree` directive.
+   You can adapt this file completely to your liking,
```
Comment (min-xu-ai) [inline docs/source/index.rst:3]: I wrapped some lines in this file.

File: fairscale/optim/__init__.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 # LICENSE file in the root directory of this source tree.
 
 """
-:mod:`fairgc.optim` is a package implementing various torch optimization algorithms.
+:mod:`fairscale.optim` is a package implementing various torch optimization algorithms.
```
Comment (min-xu-ai) [inline fairscale/optim/__init__.py:7]: I assumed that fairgc is an old name.

**Review by blefaudeux — APPROVED** [review]
LGTM, thanks for the cleanup !

**Pull request merged**